web/src/services/state/index.ts

```diff
@@ -9,8 +10,21 @@ export type SelectedWidget = {
   propertyId: string;
 };
 
+export type SelectedLayer = {
+  id: string;
+  layerId: string;
+  featureId?: string;
+  layerSelectionReason?: LayerSelectionReason;
+};
+
 export const selectedWidgetVar = makeVar<SelectedWidget | undefined>(undefined);
 
 export const selectedWidgetAreaVar = makeVar<WidgetAreaType | undefined>(undefined);
 
+export const selectedLayerVar = makeVar<SelectedLayer | undefined>(undefined);
+
+export const showDataSourceManagerVar = makeVar<boolean>(false);
+
+export const showPopOverLayerButtonVar = makeVar<boolean>(true);
+
```

Comment:
I've said this elsewhere about the showDataSouceManagerVar, but these maybe don't need to be global, so please move them more locally and pass down as props 🙏🏻

Reply:
Done.